specs/agents/tracing-instrumentation-azure.md

```diff
@@ -0,0 +1,412 @@
+
+| URL | HTTP verb | HTTP headers | HTTP query string | Resulting Operation Name |
+| --- | --------- | ---------- | ------------------- | ------------------------ |
+| | PUT    |            |   `comp=properties`                  | SetProperties    |
+| | GET    |            |   `comp=properties`                  | GetProperties    |
+| | GET    |            |   `comp=stats`                       | Stats            |
+| ends with /Tables | GET    |            |                  | Query           |
+| ends with /Tables | POST    |            |                 | Create           |
+| ends with /Tables('`<table>`') | DELETE    |            |  | Delete           |
+| ends with /`<table>` | OPTIONS    |            |           | Preflight        |
+| | HEAD      |            | `comp=acl`          | GetAcl                   |
+| | GET      |            | `comp=acl`          | GetAcl                   |
+| | PUT      |            | `comp=acl`          | SetAcl                   |
+| ends with /`<table>`() or /`<table>`(PartitionKey='`<partition-key>`',RowKey='`<row-key>`')`  | GET    |            |                  | Query            |
+| ends with /`<table>` | POST      |            |           | Insert                   |
+| ends with /`<table>`(PartitionKey='`<partition-key>`',RowKey='`<row-key>`')` | PUT      |            |           | Update                   |
+| ends with /`<table>`(PartitionKey='`<partition-key>`',RowKey='`<row-key>`')` | MERGE      |            |           | Merge                   |
+| ends with /`<table>`(PartitionKey='`<partition-key>`',RowKey='`<row-key>`')` | DELETE      |            |           | Delete                   |
+
+### File share storage
+
+Azure file service (known also as AFS or file share storage) is a managed file share service
+that uses Server Message Block (SMB) protocol or Network File System (NFS) protocol
+to provide general purpose file shares. File shares can be mounted concurrently by
+machines in the cloud or on-premises.
+
+The `<ResourceName>` is determined from the path of the URL.
+
+| APM field | Required? | Format | Notes | Example |
+| --------- | --------- | ------ | ----- | ------- |
+| `span.name` | yes | `AzureFile <OperationName> <ResourceName>` | Pascal case Operation name | `AzureFile Create directoryname` |
+| `span.type` | yes | `storage` | | |
+| `span.subtype` | yes | `azurefile` | | |
+| `span.action` | yes | `<OperationName>` | Pascal case | `Insert` |
+
+#### Span context fields
+
+| APM field | Required? | Format | Notes | Example |
+| --------- | --------- | ------ | ----- | ------- |
+| `context.destination.address` | yes | URL scheme and host | | `https://accountname.file.core.windows.net/` |
+| `context.destination.service.name` | yes | `azurefile` | | | 
+| `context.destination.service.resource` | yes | `azurefile/<ResourceName>` | | `azurefile/foo` |

```

Comment:
A storage account can contain tables, blobs, queues and file shares, so this might be too high level, unless combined with other descriptors e.g. `azure/${storage account name}/tables`, `azure/${storage account name}/blobs`, etc?

Reply:
I was thinking `azurefile/${storage account name}` for file share storage

Reply:
updated to use `azurefile/${storage account name}`